src/frontend/contexts/DraftObservationContext.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+
+import {useStore} from 'zustand';
+import CheapRuler from 'cheap-ruler';
+import * as Location from 'expo-location';
+import {AppState, type AppStateStatus} from 'react-native';
+import {
+  DraftState,
+  DraftObservationStore,
+  convertPosition,
+} from './PersistedStores/DraftObservationStore.ts';
+
+export function useDraftObservationInstance(): DraftState;
+export function useDraftObservationInstance<T>(
+  selector: (state: DraftState) => T,
+): T;
+export function useDraftObservationInstance<T>(
```

Comment:
Naming: As I understand it, you are using `instance` to refer to an instance of the store. What this hook is returning is the `state` of the store, not actually a store instance. I know the Zustand hook is called `useStore()`, but it does take a store instance as the parameter, so it makes a bit more sense. I don't feel strongly about this, feel free to ignore.

Reply:
personally would find it a bit confusing to keep the `instance` naming here given that it returns the state of the store instance and not the instance itself. It's helpful to clearly distinguish between a store instance and its state given the updated approach to zustand we're trying to adopt.

Would prefer renaming this to avoid the usage of `instance` to avoid confusion e.g. `useDraftObservation()` or `useDraftObservationState()`.